### PR TITLE
Angular v2 sample events

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds events to the Angular v2 sample, creates a `MsalBroadcastService`, allows the `MsalService` to broadcast events, and the app to subscribe to events that happen because MSAL apis were invoked. 

This fixes the login redirect state change issue in issue #2268, and addresses considerations raised in #1244.

### CODE REVIEW

File: samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.broadcast.service.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { MsalBroadcastEvent } from './constants';
+import { AuthenticationResult, AuthError } from '@azure/msal-browser';
+
+export type MessageCallback = (payload: any) => void;
```
Comment (jasonnutter) [inline samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.broadcast.service.ts:6]: I think this can be deleted

Comment (jo-arroyo) [reply samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.broadcast.service.ts]: Done!

**Comment by jo-arroyo** [comment]
@rellis-of-rhindleton Thanks again for your feedback in #1244. We took it into account here. Please let us know what you think of `MsalBroadcastService`'s use of observables.

File: samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
   ) {}
+
   ngOnInit(): void {
     this.isIframe = window !== window.parent && !window.opener;
 
     this.checkAccount();
+
+    this.broadcastService.msalSubject$
+      .pipe(
+        filter(msg => msg.type === MsalBroadcastEvent.LOGIN_SUCCESS || msg.type === MsalBroadcastEvent.ACQUIRE_TOKEN_SUCCESS),
```
Comment (jasonnutter) [inline samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts:32]: Don't have to do this here, but in the library we might consider providing convenience functions that do this filtering for each event type.

File: samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.service.ts [hunk 3]
```diff
@@ -48,10 +71,35 @@ export class MsalService implements IMsalService {
...
+                    if (authResponse) {
+                        const loggedInAccount = loggedInAccounts.find((account) => account.username === authResponse.account.username);
+                        if (loggedInAccount) {
+                            this.broadcastService.broadcast(MsalBroadcastEvent.ACQUIRE_TOKEN_SUCCESS, authResponse);
+                        } else {
+                            this.broadcastService.broadcast(MsalBroadcastEvent.LOGIN_SUCCESS, authResponse);
+                        }
+                    }
+                    return authResponse;
+                })
```
Comment (jasonnutter) [inline samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.service.ts:87]: Can you add events for `.catch`?

**Review by jasonnutter — CHANGES_REQUESTED** [review]
Looks

File: samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts (hunk 2 above)
Comment (jo-arroyo) [reply samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts:32]: Noted!

File: samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.service.ts (hunk 3 above)
Comment (jo-arroyo) [reply samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.service.ts:87]: Done.

File: samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts (hunk 2 above)
Comment (rellis-of-rhindleton) [reply samples/msal-angular-v2-samples/angular10-browser-sample/src/app/app.component.ts:32]: I don't know how fancy you want to get here, but I often go the type-safe-enum-ish route with stuff like this.

In constants.ts:
```ts
export enum MsalBroadcastEventType {
    LOGIN_SUCCESS = "msal:loginSuccess",
    LOGIN_FAILURE = "msal:loginFailure",
    ACQUIRE_TOKEN_SUCCESS = "msal:acquireTokenSuccess",
    ACQUIRE_TOKEN_FAILURE = "msal:acquireTokenFailure",
    SSO_SILENT_SUCCESS = "msal:ssoSilentSuccess",
    SSO_SILENT_FAILURE = "msal:ssoSilentFailure"
}

export interface MsalBroadcastEvent {
    eventType: MsalBroadcastEventType;
    accountChanged: boolean;
    // ...whatever else comes to mind, can be added to as needed
}

export const msalBroadcastEvents: { [key: string]: MsalBroadcastEvent } = {
    loginSucess: {
        eventType: MsalBroadcastEventType.LOGIN_SUCCESS,
        accountChanged: true
    },
    loginFailure: {
        eventType: MsalBroadcastEventType.LOGIN_FAILURE,
        accountChanged: false, // (would this be false?)
    },
    acquireTokenSuccess: {
        eventType: MsalBroadcastEventType.ACQUIRE_TOKEN_SUCCESS,
        accountChanged: true
    },
    // ...
};
```

All that ceremony just so you can do this in AppComponent:
```ts
this.broadcastService.msalSubject$
    .pipe(
        filter(msg => msg.type.accountChanged),
        takeUntil(this._destroying$)
    )
    .subscribe(msg => {
        this.checkAccount();
    });
```
I haven't given the names much thought. Convenience functions might well be easier anyway.

**Comment by rellis-of-rhindleton** [comment]
It looks good to me. Thanks for doing it.

You may want to make `msalSubject$` a get-only property so people don't shoot themselves in the foot, but I guess that isn't likely.

**Review by jmckennon — APPROVED** [review]
LGTM!

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**